common/threadloop.hpp

```diff
@@ -81,9 +85,10 @@ class threadloop : public plugin {
 		record_coalescer it_log {record_logger_};
 		std::cout << "thread," << std::this_thread::get_id() << ",threadloop," << name << std::endl;
 
+		_m_stoplight->wait_for_ready();
 		_p_thread_setup();
 
-		_m_stoplight->wait_for_ready();
+		// _m_stoplight->wait_for_ready();
 		while (!_m_stoplight->check_should_stop()) {
```

Comment:
```suggestion
		while (!_m_stoplight->check_should_stop() || should_terminate()) {
```

It seems the class already has a variable, `_m_should_terminate`, that we can use instead of `_m_internal_stop`. Fewer variables/methods is preferable. I don't _think_ it's used anywhere else.

Reply:
`_m_should_terminate` is used in [runtime/main.cpp] (https://github.com/ILLIXR/ILLIXR/blob/823e3a6161de588468244e4f9a24c86df0ebad30/runtime/main.cpp#L58)

Reply:
That is a different `_m_should_terminate` defined [here](https://github.com/ILLIXR/ILLIXR/blob/823e3a6161de588468244e4f9a24c86df0ebad30/runtime/main.cpp#L64).

Reply:
Yeah you are right

Reply:
Can I stick with `_m_internal_stop`? The name makes more sense to express what it is used for.